File: ruby/sieve/sieve.rb
class Sieve
  attr_reader :num

  def initialize(num)
    @num = num
  end

  def primes
    range = [*2..num]

    catch(:stop) do
      range.each do |num|
        original_num = num
        first_count = range.count
        until num > range[-1]
          num += original_num
          range.delete(num)
        end
        second_count = range.count
        throw :stop if first_count == second_count
      end
    end
    range
  end

end

File: javascript/sieve/sieve.js
var Sieve = function(num){
  this.primes = findPrimes(num);
};

function findPrimes(num){
  var possiblePrimes = generateNumArray(num);
  for (var i = 0; i < possiblePrimes.length; i++){
    for (var x = i+1; x < possiblePrimes.length; x ++){
      if (possiblePrimes[x] % possiblePrimes[i] === 0){
        possiblePrimes.splice(x,1);
        x -= 1;
      }
    }
  }
  return possiblePrimes;
}

function generateNumArray(num){
  var array = [];
  for(var i = 2; i <= num; i ++){
    array.push(i);
  }
  return array;
}

module.exports = Sieve;
